runtime/Builtins/Array.h

```diff
@@ -362,7 +362,7 @@ class ArraySlice {
 
     ArrayIterator<T> iterator() const
     {
-        return ArrayIterator<T> { *m_storage, m_offset, m_size };
+        return ArrayIterator<T> { *m_storage, m_offset, size() };
```

Comment:
is this because m_size doesn't encode that there is no limit to the range on that side?

Reply:
Yeah, I guess the larger question would be:
```
mut arr = [1, 2, 3, 4]
let slice = arr[..]
arr.push(5)
```
Should `slice` in that case be `[1, 2, 3, 4, 5]`
If yes I think it is correct after this fix.
If no, then it would be best to check the bounds when first setting `m_size`.

Reply:
imho, slice should be viewing `[1, 2, 3, 4]`. I think of it as a view, and the indices get filled in when you create the view.

Reply:
Yeah, I was thinking about this some more and agree. I will update it.

Reply:
This is updated now.